### PR TITLE
New Web Installer

### PR DESCRIPTION
**Pull request by murrant**
New style
Ability to go back to previous steps
Modular for use with docker and vm images
Remove last usage of PHP session
Manual entry of install with INSTALL env.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: html/install.php [hunk 1]
```diff
@@ -87,17 +80,15 @@
...
+        session()->flush();
     }
 }
+session()->save();
 
 if (empty($stage)) {
     $stage = 0;
 }
-
+var_dump(cookie());
```
Comment (Jellyfrog) [inline html/install.php:91]: :)

File: html/install.php [hunk 2]
```diff
@@ -2,60 +2,55 @@
 use LibreNMS\Authentication\LegacyAuth;
```
Comment (Jellyfrog) [inline html/install.php:2]: Shouldn't this file be deleted?

File: html/install.php (hunk 1 above)
Comment (murrant) [reply html/install.php:91]: oh, I need to delete that file :)

File: app/Http/Controllers/Install/FinalizeController.php [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    private function getConfigFileContents()
+    {
+        $db = session('db');
+        $install_dir = base_path();
+
+        return <<<"EOD"
+## Have a look in defaults.inc.php for examples of settings you can set here. DO NOT EDIT defaults.inc.php!
+
+### Database config
+\$config['db_host'] = '{$db['host']}';
```
Comment (Jellyfrog) [inline app/Http/Controllers/Install/FinalizeController.php:134]: Do we actually still need this if we set `.env`?

Comment (murrant) [reply app/Http/Controllers/Install/FinalizeController.php:134]: Not really, but I didn't want to combine this with my PR that removes that requirement.  They are a little inter-dependent though.

Comment (murrant) [reply app/Http/Controllers/Install/FinalizeController.php:134]: Maybe I just remove the install modifications from that PR...

Comment (Jellyfrog) [reply app/Http/Controllers/Install/FinalizeController.php:134]: Wouldn't it be nice with a fresh installer PR? :D

Comment (murrant) [reply app/Http/Controllers/Install/FinalizeController.php:134]: I could merge them.

**Comment by murrant** [comment]
@librenms/reviewers ready for review.  I've tested this pretty well.

File: config/database.php [hunk 4]
```diff
@@ -56,13 +53,12 @@
...
             'driver' => 'mysql',
-            'url' => env('DATABASE_URL'),
-            'host' => env('DB_HOST', $fallback_db_config['db_host']),
-            'port' => env('DB_PORT', $fallback_db_config['db_port']),
-            'database' => env('DB_DATABASE', $fallback_db_config['db_name']),
-            'username' => env('DB_USERNAME', $fallback_db_config['db_user']),
-            'password' => env('DB_PASSWORD', $fallback_db_config['db_pass']),
-            'unix_socket' => env('DB_SOCKET', $fallback_db_config['db_socket']),
+            'host' => env('DB_HOST', 'localhost'),
+            'port' => env('DB_PORT', ''),
```
Comment (Jellyfrog) [inline config/database.php:57]:
```suggestion
            'port' => env('DB_PORT', '3306'),
```

Comment (murrant) [reply config/database.php:57]: Should be handled by the underlying mysql library, but that is what upstream has, so should be fine.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-65-release-changelog-june-2020/12687/1

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-65-release-changelog-june-2020/12687/1

**Pull request merged**